client/src/components/RegisterModal.vue

```diff
@@ -0,0 +1,266 @@
+<script setup lang="ts">
+import { ref, defineEmits } from 'vue'
+import { useDisplay } from 'vuetify'
+
+const { xs } = useDisplay()
+
+const formData = ref({
+  username: '',
+  email: '',
+  firstName: '',
+  lastName: '',
+  dateOfBirth: '',
+  genderId: null,
+  indigenousTIS: null,
+  password: '',
+  confirmPassword: '',
+})
+
+const dialog = ref(false)
+const emit = defineEmits(['close'])
+
+const openDialog = () => {
+  dialog.value = true
+}
+
+const closeDialog = () => {
+  dialog.value = false
+  emit('close')
+}
+
+const submitForm = () => {
+  closeDialog()
+}
+</script>
+
+<template>
+  <div>
+    <v-btn
+      id="register-button"
+      class="bg-primaryBlue text-creamyWhite d-flex justify-center align-center"
+      @click="openDialog"
+      text="Registration"
+    ></v-btn>
+
+    <v-dialog
+      v-model="dialog"
```

Comment:
We've now got a modal store which houses the state of which modal is open. That is in ``stores/modal.ts``. So the ``v-model`` should be bound to the state housed in the store instead.